File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }

        unordered_map<char, char> s_to_t;
        unordered_map<char, char> t_to_s;
        for (int i = 0; i < s.size(); i++) {
            if (s_to_t.find(s[i]) == s_to_t.end()) {
                s_to_t[s[i]] = t[i];
            } else if (s_to_t[s[i]] != t[i]) {
                return false;
            }

            if (t_to_s.find(t[i]) == t_to_s.end()) {
                t_to_s[t[i]] = s[i];
            } else if (t_to_s[t[i]] != s[i]) {
                return false;
            }
        }

        return true;
    }
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        s_to_t = {}
        t_to_s = {}
        for i in range(len(s)):
            if not s[i] in s_to_t:
                s_to_t[s[i]] = t[i]
            elif s_to_t[s[i]] != t[i]:
                return False
            
            if not t[i] in t_to_s:
                t_to_s[t[i]] = s[i]
            elif t_to_s[t[i]] != s[i]:
                return False
            
        return True
        